README.md

```diff
@@ -2,37 +2,109 @@
 
 ## About
 
-`um2nc-standalone` is an [ACCESS-NRI](https://www.access-nri.org.au/) project to merge multiple versions of unified model conversion tools to a single, canonical project for the ESM1.5 model. 
+`um2nc-standalone` is a `Python3` utility to convert Unified Model data files to NetCDF format. 
+
+The `um2nc-standalone` project is an [ACCESS-NRI](https://www.access-nri.org.au/) initiative to merge multiple versions of unified model conversion tools to a single, canonical project for the ESM1.5 model. 
 
 ## Installation
 
-TODO
+These installation instructions assume deployment to a `Linux` or `MacOS` platform.
 
-* `virtualenv` instructions
-* `conda`/`miniconda`/`micromamba?` instructions
+`Windows` is currently **unsupported**.
 
-## User documentation
+### Downloading `um2netcdf-standalone`
 
-TODO
+```commandline
+cd <your projects dir>
+git clone https://github.com/ACCESS-NRI/um2nc-standalone.git
+cd um2nc-standalone
+```
+
+### Installing Dependencies
+
+#### MacOS
+
+```Bash
+$ brew install udunits
+```
+
+if this fails with some compiler/header errors, try:
+
+```Bash
+# set up non standard cf-unit paths
+$ export UDUNITS2_XML_PATH=/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2.xml
+$ export UDUNITS2_LIBDIR=/opt/homebrew/lib
+$ export UDUNITS2_INCDIR=/opt/homebrew/Cellar/udunits/2.2.28/include
+$ brew install udunits
+```
+
+#### Linux
+
+```Bash
+$ apt-get install udunits
+```
+
+**TODO:** add `yum` commands?
+
+## Creating a Python Runtime Environment
+
+The following instructions outline two options for creating a Python runtime environment fot `um2netcdf-standalone`. The purpose of a runtime environment is to manage specific library dependencies, separately to the system python.
+
+`virtualenv` is a low level python environment manager, with a companion `virtualenvwrapper` project to provide greater usability. These packages are available in `apt` for Linux and `homebrew` on Mac. The following instructions assume use of `virtualenvwrapper` ([virtualenvwrapper docs](https://virtualenvwrapper.readthedocs.io/en/latest/)).
+
+`conda` and related tools such as `miniconda` & `micromamba` are higher level environment managers than `virtualenv.
+
+If you are unsure which environment manager to use and/or new to Python, `conda` is the recommended approach. 
+
+### Creating a `virtualenv` environment
+
+```Bash
+# assuming virtualenvwrapper has been installed & configured on your system...
+# e.g. "brew install virtualenvwrapper" or "apt-get install virtualenvwrapper"
+$ mkvirtualenv -r requirements.txt ums
+
+# should download & install multiple packages
+# the command prompt should change with a (ums) prefix 
+```
+
+### Creating a `conda` environment
+
+```Bash
+# works for Linux and MacOS
+$ conda create -n ums
+$ activate ums
+$ conda install pip
+$ 
+$ cd <your-um2netcdf-project-dir>
+$ pip install -r requirements.txt
```

Comment:
Which requirements are requiring us to use `pip install` (as opposed to `conda install`)?

Reply:
Ooops, this might be a cut/paste/hasty typing error. I _barely_ use `conda`, but I think this command is supposed to be `conda install --file requirements.txt`.
Are you a regular `conda` user?

Reply:
Not a super-user, but I regularly try to stay away from pip if at all possible (conda gives better compartmentalization). The `requirements.txt` given won't play nice with conda due to some missing packages. We might consider setting up an `environment.yaml` instead that describes the required packages (from memory, that allows you to explicitly set which packages need to come in via pip).

Reply:
Quick note: `requirements.txt` came from `pip freeze` from my local virtualenv (the get something documented 1st approach).

Reply:
For what it's worth, the install "docs" were ripped from https://github.com/ACCESS-NRI/amami/issues/13. The amami project included a version of `um2nc`, so the virtualenv dependencies are similar or the same.

From what I recall, getting an environment built using that method required the hacky `conda` & `pip` commands...

Reply:
I'm still mulling the `virtualenv` vs `conda` question. Should we even be supplying/documenting a `virtualenv` install option? I think the world has pretty definitively shifted over to `conda` now (and besides, everything else I've seen so far at ACCESS-NRI appears to stick with `conda`).

Reply:
> Should we even be supplying/documenting a `virtualenv` install option? I think the world has pretty definitively shifted over to `conda` now (and besides, everything else I've seen so far at ACCESS-NRI appears to stick with `conda`).

I added `virtualenv` instructions because of the... _old habit_ of using lower level tools locally for yeeeears.

There should be a working `conda` setup config via the newly added CI content. The docs could be rejigged to focus mainly on `conda`, assuming most folks are non-developers, simply needing to setup & run the code. `virtualenv` instructions could be hived off to a separate section for those needing to create an isolated Py environment. I don't know who the users are, so I'm unsure of the most useful approach to take.

Reply:
I just discovered the `pipreqs` package. Running it over the code gives the following (much reduced) dependency list:

```
cf_units==3.2.0
cftime==1.6.4
f90nml==1.4.4
mule==2020.1.1
netCDF4==1.7.1
netCDF4==1.7.1.post2
numpy==2.1.1
pytest==8.3.2
scitools_iris==3.10.0
```
I'm pretty sure the versions shown correspond to what is currently installed in my environment, not what is actually required. All of these except `cf_units` and `scitools_iris` can be installed via `conda`; those two require `pip`.

I'm still having trouble making the tests run to see if this new 'minimal' environment would work, I managed to blow away my package import updates doing other things...

Reply:
Got my imports to work! And confirmed that a bare conda environment with only the above requirements explicitly installed passes the test suite.